modules/ClinGen.py

```diff
@@ -1,137 +1,163 @@
-
-        if len(self.unmapped_diseases) > 0:
-            logging.info(f'There are {len(self.unmapped_diseases)} unmapped diseases.')
-            unmapped_diseases_string = "\n- ".join([x[1] for x in self.unmapped_diseases])
-            logging.info(f'Unmapped diseases: \n- {unmapped_diseases_string}')
-
-    def write_evidence_strings(self, filename):
-        logging.info('Writing ClinGen evidence strings to %s', filename)
-        with gzip.open(filename, 'wt') as tp_file:
-            for evidence_string in self.evidence_strings:
-                json.dump(evidence_string, tp_file)
-                tp_file.write('\n')
-
-
-def main(infile, outfile):
-
-    clingen = ClinGen()
-    clingen.process_gene_validity_curations(infile, outfile)
+import logging
+import os
+import tempfile
+
+from pyspark.conf import SparkConf
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.functions import array, col, first, lit, monotonically_increasing_id, struct, trim
+from pyspark.sql.types import StringType, StructType, TimestampType
+
+from common.ontology import add_efo_mapping
+
+
+def main(input_file: str, output_file: str, cache_dir: str, local: bool = False) -> None:
+
+    # Initialize spark session
+    if local:
+        sparkConf = (
+            SparkConf()
+            .set('spark.driver.memory', '15g')
+            .set('spark.executor.memory', '15g')
+            .set('spark.driver.maxResultSize', '0')
+            .set('spark.debug.maxToStringFields', '2000')
+            .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
+        )
+        spark = (
+            SparkSession.builder
+            .config(conf=sparkConf)
+            .master('local[*]')
+            .getOrCreate()
+        )
+    else:
+        sparkConf = (
+            SparkConf()
+            .set('spark.driver.maxResultSize', '0')
+            .set('spark.debug.maxToStringFields', '2000')
+            .set('spark.sql.execution.arrow.maxRecordsPerBatch', '500000')
+        )
+        spark = (
+            SparkSession.builder
+            .config(conf=sparkConf)
+            .getOrCreate()
+        )
+
+    # Read and process Clingen's table into evidence strings
+
+    clingen_df = read_input_file(input_file, spark_instance=spark)
+    logging.info('Gene Validity Curations table has been imported. Processing evidence strings.')
+
+    evidence_df = process_clingen(clingen_df)
+
+    evidence_df = add_efo_mapping(evidence_strings=evidence_df, spark_instance=spark, ontoma_cache_dir=cache_dir)
+    logging.info('Disease mappings have been added.')
+
+    write_evidence_strings(evidence_df, output_file)
+    logging.info(f'{evidence_df.count()} evidence strings have been saved to {output_file}')
+
+def read_input_file(input_file: str, spark_instance) -> DataFrame:
+    '''
+    Reads Gene Validity Curations CSV file into a Spark DataFrame forcing the schema
+    The first 6 rows of this file include metadata that needs to be dropped
+    '''
+
```

Comment:
Docstring as per PEP8

```suggestion
    """
    Reads Gene Validity Curations CSV file into a Spark DataFrame forcing the schema
    The first 6 rows of this file include metadata that needs to be dropped
    """

```